lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -183,11 +185,20 @@ export class PublicClientApplication implements IPublicClientApplication {
             return null;
         }
 
-        // Hash contains known properties - handle and return in callback
-        const currentAuthority = this.browserStorage.getCachedAuthority();
-        const authClient = await this.createAuthCodeClient(currentAuthority);
-        const interactionHandler = new RedirectHandler(authClient, this.browserStorage);
-        return interactionHandler.handleCodeResponse(responseHash, this.browserCrypto);
+        const correlationId = this.browserStorage.getItem(this.browserStorage.generateCacheKey(TemporaryCacheKeys.CORRELATION_ID), CacheSchemaType.TEMPORARY) as string;
```

Comment:
This item is already available as part of the cached auth code request, might be easier to get it from the request object?